src/cobra/manipulation/validate.py

```diff
@@ -12,23 +17,48 @@
 }
 
 
-def check_mass_balance(model):
+def check_mass_balance(model: "Model") -> Dict["Reaction", Dict["Metabolite", float]]:
+    """Check mass balance for reactions of `model` and return unbalanced ones.
+
+    Parameters
+    ----------
+    model: cobra.Model
+        The model to perform check on.
+
+    Returns
+    -------
+    dict of {cobra.Reaction: dict of {cobra.Metabolite: float}}
+        Returns an empty dict if all components are balanced.
+
+    """
     unbalanced = {}
     for reaction in model.reactions:
-        if reaction.annotation.get("sbo") not in NOT_MASS_BALANCED_TERMS:
+        if reaction.annotation.get("sbo") not in _NOT_MASS_BALANCED_TERMS:
             balance = reaction.check_mass_balance()
             if balance:
                 unbalanced[reaction] = balance
     return unbalanced
 
 
-def check_metabolite_compartment_formula(model):
+def check_metabolite_compartment_formula(model: "Model") -> List[str]:
```

Comment:
This only checks the formula. No idea why the compartment is mentioned in the function name. But this should at least be corrected in the Dichtring.

Reply:
Yes you are right. I will replace the `compartments` by `formulae`.